compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -6412,8 +6412,16 @@ bool directMemoryStoreHelper(TR::CodeGenerator* cg, TR::Node* storeNode)
                TR::DebugCounter::incStaticDebugCounter(cg->comp(), "z/optimization/directMemoryStore/indirect");
 
                // Force the memory references to not use an index register because MVC is an SS instruction
+               // After generating a memory reference, Enforce it to generate LA instruction.
+               // This will avoid scenarios when we have common base/index between destination and source
+               // And when generating the source memory reference, it clobber evaluates one of the node shared between 
+               // target memory reference as well.
+               // TODO : Fix this. It should not clobber evaluate the node in first place. Workaround just suppresses the bug.
```

Comment:
In general we shouldn't be adding TODO's to the code because they will simply be forgotten.  Is there an issue open to track the problem this fix is working around?  If not, please create one and reference this PR.

Does this PR have any value once that overarching issue is fixed?